hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/BackendEntityEnricher.java

```diff
@@ -72,6 +78,14 @@ public void enrichTrace(StructuredTrace trace) {
         .forEach(pair -> decorateWithBackendEntity(pair.getRight().get(), pair.getLeft(), trace));
   }
 
+  @Override
+  public void enrichEvent(StructuredTrace trace, Event event) {
```

Comment:
Based on the current definition, we should add new attributes as part of this method - `decorateWithBackendEntity`.

We don't have to enrich all the spans/events with backend_operation.

Reply:
I was going to ask that.

Do we want to add the db operation and other attributes as part of backend entity?

If yes, then we need to take care of the backend entity update, if there is any change in the attributes. The BackendEntityEnricher currently just creates a backend entity and never updates it, which I believe is a wrong behavior, because I see many attributes in backend entity, which should have been updated

Reply:
> Do we want to add the db operation and other attributes as part of backend entity?

No. But operation attributes depends on the kind of backend. So, in the current setup, we need a way to pass such additional attributes to `decorateWithBackendEntity` where we are enriching event.